components/formats-gpl/src/loci/formats/in/HarmonyReader.java

```diff
@@ -0,0 +1,783 @@
+        for (int field=0; field<fields.length; field++) {
+          int nextPlane = 0;
+          for (int t=0; t<ts.length; t++) {
+            for (int z=0; z<zs.length; z++) {
+              for (int c=0; c<cs.length; c++) {
+                for (Plane p : planeList) {
+                  if (p.row == rows[row] && p.col == cols[col] &&
+                    p.field == fields[field] && p.t == ts[t] && p.z == zs[z] &&
+                    p.c == cs[c])
+                  {
+                    planes[nextSeries][nextPlane] = p;
+                    break;
+                  }
+                }
+                nextPlane++;
+              }
+            }
+          }
+          nextSeries++;
+        }
+      }
+    }
+
+    reader = new MinimalTiffReader();
+
+    for (int i=0; i<seriesCount; i++) {
+      CoreMetadata ms = new CoreMetadata();
+      core.add(ms);
+      ms.sizeX = planes[i][0].x;
+      ms.sizeY = planes[i][0].y;
+      ms.sizeZ = uniqueZs.size();
+      ms.sizeC = uniqueCs.size();
+      ms.sizeT = uniqueTs.size();
+      ms.dimensionOrder = "XYCZT";
+      ms.rgb = false;
+      ms.imageCount = getSizeZ() * getSizeC() * getSizeT();
+
+      RandomAccessInputStream s =
+        new RandomAccessInputStream(planes[i][0].filename, 16);
+      TiffParser parser = new TiffParser(s);
+      parser.setDoCaching(false);
+
+      IFD firstIFD = parser.getFirstIFD();
+      ms.littleEndian = firstIFD.isLittleEndian();
+      ms.pixelType = firstIFD.getPixelType();
+      s.close();
+    }
+
+    HashMap<String, String> xmlMetadata = handler.getMetadataMap();
+    for (String key : xmlMetadata.keySet()) {
+      addGlobalMeta(key, xmlMetadata.get(key));
+    }
+
+    // populate the MetadataStore
+
+    MetadataStore store = makeFilterMetadata();
+    MetadataTools.populatePixels(store, this, true);
+
+    store.setPlateID(MetadataTools.createLSID("Plate", 0), 0);
+    store.setPlateRows(new PositiveInteger(handler.getPlateRows()), 0);
+    store.setPlateColumns(new PositiveInteger(handler.getPlateColumns()), 0);
+
+    String plateAcqID = MetadataTools.createLSID("PlateAcquisition", 0, 0);
+    store.setPlateAcquisitionID(plateAcqID, 0, 0);
+
+    PositiveInteger fieldCount = FormatTools.getMaxFieldCount(fields.length);
+    if (fieldCount != null) {
+      store.setPlateAcquisitionMaximumFieldCount(fieldCount, 0, 0);
+    }
+    String startTime = handler.getMeasurementTime();
+    if (startTime != null) {
+      store.setPlateAcquisitionStartTime(new Timestamp(startTime), 0, 0);
+    }
+
+    for (int row=0; row<rows.length; row++) {
+      for (int col=0; col<cols.length; col++) {
+        int well = row * cols.length + col;
+        store.setWellID(MetadataTools.createLSID("Well", 0, well), 0, well);
+        store.setWellRow(new NonNegativeInteger(rows[row]), 0, well);
+        store.setWellColumn(new NonNegativeInteger(cols[col]), 0, well);
+
+        for (int field=0; field<fields.length; field++) {
```

Comment:
This looks to me like each well sample gets the same number of fields. This is not necessarily the case, in particular in the future versions of Harmony datasets will have different numbers of fields. Is this something that is supported?